modules/reitit-core/src/reitit/coercion.cljc

```diff
@@ -98,17 +111,24 @@
 (defn extract-response-format-default [request _]
   (-> request :muuntaja/response :format))
 
-(defn response-coercer [coercion body {:keys [extract-response-format serialize-failed-result]
-                                       :or {extract-response-format extract-response-format-default}}]
+(defn response-coercer [coercion {:keys [content body]} {:keys [extract-response-format serialize-failed-result]
+                                                         :or {extract-response-format extract-response-format-default}}]
   (if coercion
-    (if-let [coercer (-response-coercer coercion body)]
-      (fn [request response]
-        (let [format (extract-response-format request response)
-              value (:body response)
-              result (coercer value format)]
-          (if (error? result)
-            (response-coercion-failed! result coercion value request response serialize-failed-result)
-            result))))))
+    (let [per-format-coercers (some->> (for [[format schema] content]
+                                         [format (-response-coercer coercion schema)])
+                                       (filter second)
+                                       (seq)
+                                       (into {}))
+          default (when body (-response-coercer coercion body))]
+      (when (or per-format-coercers default)
+        (fn [request response]
+          (let [format (extract-response-format request response)
+                value (:body response)
+                coercer (get per-format-coercers format (or default (fn [value _format] value)))
```

Comment:
the fn `(fn [value _format] value)` is used in both request & response coercion and is generated per each request, could be extracted into a named fn in this ns.

Reply:
done